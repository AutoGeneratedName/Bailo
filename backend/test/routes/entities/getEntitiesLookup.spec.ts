import { describe, expect, test, vi } from 'vitest'

import { testGet } from '../../testUtils/routes.js'

vi.mock('../../../src/utils/user.js')

const authenticationMocks = vi.hoisted(() => ({
  authenticationMiddleware: vi.fn(() => []),
  getMultipleUsersInformation: vi.fn(() => [
    {
      email: `joeb@example.com`,
      name: 'Joe Bloggs',
      organisation: 'Acme Corp',
    },
    {
      email: `janeb@example.com`,
      name: 'Jane Bloggs',
      organisation: 'Acme Corp',
    },
  ]),
}))
vi.mock('../../../src/connectors/authentication/index.js', async () => ({
  default: authenticationMocks,
}))

vi.mock('../../../src/utils/entity.js', async () => ({
  toEntity: vi.fn(() => 'user:userdn'),
}))

describe('routes > entities > getEntitiesLookup', () => {
  test('200 > ok', async () => {
    const res = await testGet(`/api/v2/entities/lookup?dnList=user1&dnList=user2`)

    expect(res.statusCode).toBe(200)
    expect(authenticationMocks.getMultipleUsersInformation.mock.calls).matchSnapshot()
    expect(res.body).matchSnapshot()
  })
})
