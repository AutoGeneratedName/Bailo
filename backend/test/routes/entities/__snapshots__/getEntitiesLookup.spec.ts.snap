// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`routes > entities > getEntitiesLookup > 200 > ok 1`] = `
[
  [
    [
      "user:userdn",
      "user:userdn",
    ],
  ],
]
`;

exports[`routes > entities > getEntitiesLookup > 200 > ok 2`] = `
{
  "entities": [
    {
      "email": "joeb@example.com",
      "name": "Joe Bloggs",
      "organisation": "Acme Corp",
    },
    {
      "email": "janeb@example.com",
      "name": "Jane Bloggs",
      "organisation": "Acme Corp",
    },
  ],
}
`;

exports[`routes > entities > getEntityLookup > 200 > ok 1`] = `
[
  [
    [
      "user:userdn",
      "user:userdn",
    ],
  ],
]
`;

exports[`routes > entities > getEntityLookup > 200 > ok 2`] = `
{
  "entities": [
    {
      "email": "joeb@example.com",
      "name": "Joe Bloggs",
      "organisation": "Acme Corp",
    },
    {
      "email": "janeb@example.com",
      "name": "Jane Bloggs",
      "organisation": "Acme Corp",
    },
  ],
}
`;
